### PR TITLE
Wishlist

### PR DESCRIPTION
**Pull request by Pratishah**
Added WishList Feature and Navbar icon for Wishlist.

Screenshot (34)
Screenshot (35)
Screenshot (36)

### CODE REVIEW

File: src/redux/WishList/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { CarDetails } from "./../../models/CarDetails";
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+let initialState: { wishCars: CarDetails[]; error: boolean } = {
+  wishCars: [],
+  error: false,
+};
+
+export const WishListSlice = createSlice({
```
Comment (pashedin) [inline src/redux/WishList/reducer.ts]: please use camelCase, starting with lower case

File: src/pages/WishList/WishList.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  const uniqueWishList: CarDetails[] = Array.from(new Set(carsWishList));
+
+  const [wishCars, setwishCars] = useState<CarDetails[]>();
+  useEffect(() => {
+    window.scrollTo(0, 0);
+    document.title = "Xtreme Cars | WishList";
+    return () => {
+      setwishCars(uniqueWishList);
+    };
```
Comment (pashedin) [inline src/pages/WishList/WishList.tsx]: why this?

File: src/pages/WishList/WishList.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import "./WishList.scss";
+
+const WishList = () => {
+  const carsWishList = useSelector(
+    (state: any) => state.wishListReducer.wishCars
+  );
+
+  const uniqueWishList: CarDetails[] = Array.from(new Set(carsWishList));
+
+  const [wishCars, setwishCars] = useState<CarDetails[]>();
```
Comment (pashedin) [inline src/pages/WishList/WishList.tsx]: camelCase for `setwishcars`

File: src/components/Navbar/Navbar.tsx [hunk 4]
```diff
@@ -4,9 +4,13 @@ import { Link, NavLink, useNavigate } from "react-router-dom";
 import Logo from "../../assets/logo.svg";
 import loggedProfile from "../../assets/man.png";
 import cartImage from "../../assets/cartImage.svg";
+import WishListIcon from "../../assets/WishlistIcon.png";
 import UnknownProfile from "../../assets/profile.svg";
 import "./Navbar.scss";
 const Navbar = () => {
+  const WishCars = useSelector((state: any) => state.wishListReducer.wishCars);
```
Comment (pashedin) [inline src/components/Navbar/Navbar.tsx]: Only React components have CamelCaseing starting with upper case, rest all should have camelCase starting with lower case

File: src/components/CarCard/CarCard.tsx [hunk 5]
```diff
@@ -1,25 +1,39 @@
...
           <img src={rightarrow} alt="detail" />
         </div>
       </div>
       <div className="car-card-btn-group">
-        <button>Add to cart</button>
-        <button> Add to compare</button>
-        <button> Add to wishlist</button>
+        <button className="car-card-btn">Add to cart</button>
+        <button className="car-card-btn"> Add to compare</button>
+        <button className="car-card-btn" onClick={() => wishListHandler()}>
```
Comment (pashedin) [inline src/components/CarCard/CarCard.tsx]:
```suggestion
        <button className="car-card-btn" onClick={wishListHandler}>
```

File: src/components/CarCard/CarCard.tsx [hunk 6]
```diff
@@ -1,25 +1,39 @@
...
+        className="car-card-car-image"
+        src={props.carData.image}
+        alt="car-img"
+      />
       <div className="car-card-data">
         <div>
-          <h6> {carData.name}</h6>
-          <p>{carData.price} kh onwards</p>
+          <h6> {props.carData.name}</h6>
+          <p>{props.carData.price} kh onwards</p>
```
Comment (pashedin) [inline src/components/CarCard/CarCard.tsx]: missing an `a` there in lakh

**Review by pashedin — CHANGES_REQUESTED** [review]

**Comment by Pratishah** [comment]
added changes for wishlist actions and deleted comments and nav buttons

**Review by pant-manas — APPROVED** [review]
@kboodi can you also review once and merge the changes to main

File: src/components/CarCard/CarCard.tsx [hunk 7]
```diff
@@ -2,30 +2,50 @@ import React, { useEffect, useState } from "react";
...
-  const addToCompareCars = ()=>{
-    dispatch(addToCompare(props.carData.id))
-  }
-  debugger
+
+  const [disable, setDisable] = useState(false);
+  const addToCompareCars = () => {
+    dispatch(addToCompare(props.carData.id));
+  };
+  debugger;
```
Comment (kboodi) [inline src/components/CarCard/CarCard.tsx]: please remove debugger

File: public/index.html [hunk 8]
```diff
@@ -2,20 +2,21 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/logo.svg" />
+    <!-- <link rel="icon" href="%PUBLIC_URL%/logo.svg" /> -->
```
Comment (kboodi) [inline public/index.html:5]: commented the title icon?

File: src/components/CarCard/CarCard.tsx [hunk 9]
```diff
@@ -40,13 +60,24 @@ const CarCard = (props: {carData: CarDetails}) => {
         </div>
```
Comment (kboodi) [inline src/components/CarCard/CarCard.tsx:59]: missing a in Lakh

**Review by kboodi — APPROVED** [review]

**Pull request merged**